dipy/workflows/base.py

```diff
@@ -60,18 +60,35 @@ def __init__(self, prog=None, usage=None, description=None, epilog=None,
         self.doc = None
 
     def add_workflow(self, workflow):
-        specs = inspect.getargspec(workflow)
-        doc = inspect.getdoc(workflow)
-        self.doc = NumpyDocString(doc)['Parameters']
+        """ Take a workflow object and use introspection to extract the parameters,
+            types and docstrings of its run method. Then add these parameters
+            to the current arparser's own params to parse. If the workflow is of
+            type combined_workflow, the optional input parameters of its
+            sub workflows will also be added.
+
+            Parameters
+            -----------
+            workflow : dipy.workflows.workflow.Workflow
+                Workflow from which to infer parameters.
+        """
+        specs = inspect.getargspec(workflow.run)
+        doc = inspect.getdoc(workflow.run)
+        npds = NumpyDocString(doc)
+        self.doc = npds['Parameters']
+        self.description = ' '.join(npds['Extended Summary'])
 
-        self.outputs = NumpyDocString(doc)['Outputs']
+        self.outputs =\
```

Comment:
I think it would read better like this:
```python
        self.outputs = [param for param in npds['Parameters']
                        if 'out_' in param[0]]
```